properly zero-initialize u3_wcon linked-list pointer

joemfb opened the pull request:
... fixing a platform-dependent crash when we traverse the list to dispose it

benjamin-tlon commented:
What's going on with this crash? Something broken in `release-candidate` that doesn't always happen?

:+1: on the change itself; though it seems like we should consider ways of making errors like this less easy to make.

joemfb commented:
It's the "hija-tcsetattr-1: Interrupted system call" printf in `term.c`. Something is intermittently wrong with tty, and in CI, that's pty.js. Long story short, I must have messed up when I extracted and refactored the test runner.

joemfb commented:
I'm restarting the build, that usually works.
